src/Native/SymSgdNative/SparseBLAS.h

```diff
@@ -0,0 +1,26 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#pragma once
+#include "../Stdafx.h"
+#include "mkl.h"
+#ifndef COMPILER_GCC
+#pragma comment(lib, "../../../Libraries/MKL/Win/Microsoft.ML.MklImports.lib")
```

Comment:
>Libraries/MKL/Win/Microsoft.ML.MklImports.lib [](start = 31, length = 45)

Hi @codemzs thanks for looking at this. We need to either shift this thing to not use MKL, or port the libraries. However, considering that this involves the static linking of the library, and all other usages are using it as a dynamic library, this becomes especially awkward here.

Reply:
I wonder if temporarily (or not so temporarily) we can shift to using regular ops rather than calling MKL, based on some compile time flag (as we do elsewhere). If on, then it will compile with MKL support. If off, it does it manually. Since these are ops like `SAXPYI` and whatnot, it's not like they would be incredibly hard to support.

Reply:
OK. Once that PR is in, assuming it works, you would add the nuget dependency in which project?

---
In reply to: [205300173](https://github.com/dotnet/machinelearning/pull/556#discussion_r205300173) [](ancestors = 205300173)

Reply:
Microsoft.ML.Tests and Microsoft.ML.Predictor.Test. Anything that uses sym sgd would need the reference to the nuget so that binaries are copied over to the project folder in bin directory.

---
In reply to: [205467167](https://github.com/dotnet/machinelearning/pull/556#discussion_r205467167) [](ancestors = 205467167,205300173)